internal/mempool/types.go

```diff
@@ -70,33 +82,145 @@ type Mempool interface {
 		newPreFn PreCheckFunc,
 		newPostFn PostCheckFunc,
 	) error
+}
 
-	// FlushAppConn flushes the mempool connection to ensure async callback calls
-	// are done, e.g. from CheckTx.
-	//
-	// NOTE:
-	// 1. Lock/Unlock must be managed by caller.
-	FlushAppConn(context.Context) error
+// OpResult is the result of a pool operation. This result informs the ABCI type
+// what happened in the storage/pool layer so it can maintain cache coherence.
+type OpResult struct {
+	AddedTXs   types.Txs
+	RemovedTXs types.Txs
+	Status     Status
+}
+
+// PoolMeta is the metadata for a given pool.
+type PoolMeta struct {
+	// Type describes the type of mempool store. Examples could be priority or narwhal.
+	Type string
+	// Size is the num of TXs or whatever the unit of measure for the store.
+	Size int
+	// TotalBytes is a measure of hte store's data size.
+	TotalBytes int64
+}
+
+// Pool defines the underlying pool storage engine behavior.
+type Pool interface {
```

Comment:
this seems to not be used anywhere else. Since we separated concerns between implementation details and other module dependents It might make sense to make this private and allow other implementations of mempools have their own pool design? If im getting this wrong then please let me know, I haven't looked at tendermint interdependence of module in a while.